src/gui/wizard/owncloudwizard.cpp

```diff
@@ -106,6 +119,11 @@ OwncloudWizard::OwncloudWizard(QWidget *parent)
     connect(this, &OwncloudWizard::onActivate, _flow2CredsPage, &Flow2AuthCredsPage::slotPollNow);
 }
 
+void OwncloudWizard::reject()
+{
+    QApplication::quit();
```

Comment:
If you don't want the `Esc` key to quit the application, you can always change what the `Esc` key does. If the issue you're concerned about is the user quitting the application by accident, then you can add a confirmation dialog. 

On the other hand, it is user-hostile behavior to make it difficult for the user to quit the application in a standard and straightforward way if they so desire. The existing problem is that if the user aborts the wizard without completing the setup process, the application process that remains active is doing nothing but pestering the user to resume setup, which is archetypal hostile behavior.

Reply:
I'm not disagreeing it should quit. I'm just wondering why it quits unconditionally. I'd expect the following to happen now:

1. I start the client for the first time
2. I create an account with the wizard which appeared
3. Time passes, days or so
4. I fire up the wizard to create a second account
5. For some reason I reject the wizard (missing info, or whatever... just changed my mind)
6. The application quits

This is what concerns me with that patch. Sorry if that was unclear.

Reply:
If you're okay with me expanding the scope slightly, I could try and implement a basic quit-confirmation.

Reply:
I think our discussions are crossing a bit between this and the corresponding issue. :-)

Anyway, I'm not sure the quit-confirmation is necessary (at least I don't see Jan asking for this). Currently my main beef with this is more the unconditional quitting for the reason I mentioned above. In my opinion you want to initiate the quitting *only* if the rejection leads to the application having no account, otherwise it's OK to reject and the application can carry on.